### PR TITLE
Bugfix/tests

### PR DESCRIPTION
**Pull request by Kick1911**
*Fixing tests to run on python 3.7*

- All tests pass on tox (besides the ones that are skipped)
- 1 test fails on `pytest --use-docker tests/` (`tests/apps/test_measure.py::test_with_orig[CSVStore]`)

### Main changes
- Tag for docker image `safarov/freeswitch` must now be specified. Using `1.10.3`

### CODE REVIEW

**Comment by goodboy** [comment]
@Kick1911 oh wow thanks for doing this :partying_face: 

I'm going to take an in depth look this afternoon EST.

Talk soon :surfer:

**Comment by goodboy** [comment]
@Kick1911 one thing that could happen right now is re-enabling the CI run in travis, though likely flipping to GH actions would be more desirable for a load heavy test set like this.

I think we disabled it as part of #65.

File: tests/test_connection.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def con(fshost, loop):
 @pytest.mark.parametrize(
     'password, expect_auth',
     [('doggy', False), ('ClueCon', True)],
-    ids=lambda item: "pw={}, expect_auth={}".format(*item),
+    # ids=lambda item: "pw={}, expect_auth={}".format(*item),
```
Comment (goodboy) [inline tests/test_connection.py:36]: Is there a reason this was removed?

It's not a big deal but iirc these `ids` tags sure are handy when doing manual runs of the test suite from console.

**Review by goodboy — APPROVED** [review]
@sudonum yeah this looks great to me :+1: 

The only thing I'd love to see in this PR is tweaks to get CI working again.
Let me see if I can figure that out this afternoon.

**Comment by goodboy** [comment]
Huh, so can't remember if I disabled the CI run or not but the results from this branch are here:
https://travis-ci.org/github/friends-of-freeswitch/switchio

Looks like for nightly we need to install a newer C++ compiler.
I'm tempted to just move everything to GH actions since it's been far superior in a lot of ways and it's more juice for parallel builds.

**Comment by goodboy** [comment]
There, travis needed to be *re-synced* :roll_eyes:

File: tests/test_connection.py (hunk 1 above)
Comment (Kick1911) [reply tests/test_connection.py]: I have added it back in, but it works differently now.

**Comment by Kick1911** [comment]
I am busy trying to get GH action to work on my fork.

**Comment by goodboy** [comment]
@Kick1911 just looking at your [actions run](https://github.com/sudonum/switchio/runs/1409347535?check_suite_focus=true) and it looks like we might be still hitting the same issue of [needing a health check directive in the dockerfile as required by the `pytest-dockerctl` plugin](https://github.com/goodboy/pytest-dockerctl#caveats).

I think this may be why the CI stopped working initially as well?
I don't recall requiring it initially and maybe it was changes to `docker-py`/`docker` that allowed this to work before?

Maybe we can just stick in a flag to ignore the health check in the plugin or see if this freeswitch image can be changed to include it?

**Comment by goodboy** [comment]
Huh, [the health check appears to be in the base image](https://github.com/signalwire/freeswitch/blob/master/docker/base_image/Dockerfile) so i'm not entirely sure what's wrong here. It also is in the [official image](https://github.com/signalwire/freeswitch/blob/master/docker/release/Dockerfile). I'm slightly confused what's wrong; maybe it's [this attr path is incorrect](https://github.com/sudonum/switchio/runs/1409347535?check_suite_focus=true#step:7:79)?

**Comment by goodboy** [comment]
@Kick1911 if you're confident in this I'm fine to merge. 

This project needs some life brought back 😂

**Comment by Kick1911** [comment]
> @Kick1911 if you're confident in this I'm fine to merge.
> 
> This project needs some life brought back

I am not completely happy with it yet. In my experimentation branch I realised that some changes I made are not backwards compatible. I realised this when manually running the tests on my laptop with different versions on python 3.x.
So, I would really like the testing to work so we could not make mistakes like that. I think I will give it another try, this time making a docker on my laptop that runs docker and runs the tests. (I was using VMs to run the tests)
Logic is if I can recreate the error on my laptop, it will be much easier to debug.

I made it backwards compatible here:
https://github.com/friends-of-freeswitch/switchio/commit/d6b56c191c26cac2e5138c213677f4bda5eb529e

**Review by moises-silva — APPROVED** [review]

**Comment by goodboy** [comment]
> I realised that some changes I made are not backwards compatible. I realised this when manually running the tests on my laptop with different versions on python 3.x.
So, I would really like the testing to work so we could not make mistakes like that. I think I will give it another try, this time making a docker on my laptop that runs docker and runs the tests.

@Kick1911 sounds good.

I'm pretty sure running the tests locally should be pretty straight forward.
As mentioned I'm pretty sure the latest issue in CI is just some silly health check thing.

**Comment by Kick1911** [comment]
Ok. Found out what the problem was by pure luck.
Added the fix for the testing problem and backwards compatibility for < python 3.7.
You can check how the tests run on our fork PR #2.
https://github.com/sudonum/switchio/pull/2/checks

All tests pass for python 3.6. One test fails for python 3.7 and 3.8.
They fail on `tests/apps/test_measure.py::test_with_orig[CSVStore] FAILED` with a TimeoutError.
I don't know what that is about. Maybe you can guide me on that.

**Comment by Kick1911** [comment]
@goodboy I think you can merge it now and enable the github action that I have added if you are happy. Unless you want to see all tests passing, then I can change it to only run tests for python 3.6 to get the nice green tick.

**Comment by goodboy** [comment]
hey @Kick1911 thanks for this work and sorry about 👻 ing again - I'm not paying attention to the GH feed as much as I should.

> You can check how the tests run on our fork PR number 2.

Amazing. nice to see the suite run on 3.8 as well.
I'm noticing that 3.6 and 3.7 runs hung and were cancelled?
Any idea what's going on there?

>  then I can change it to only run tests for python 3.6 to get the nice green tick.
Weird i only see the 3.8 run actually terminating with the one error.

> They fail on tests/apps/test_measure.py::test_with_orig[CSVStore] FAILED with a TimeoutError.

Can we maybe just skip this test for now and make an issue about it?

Thanks again for all this hard work!

File: tests/conftest.py [hunk 2]
```diff
@@ -67,7 +67,7 @@ def containers(request, confdir):
     if request.config.option.usedocker:
         docker = request.getfixturevalue('dockerctl')
         with docker.run(
-            'safarov/freeswitch',
+            'safarov/freeswitch:1.10.3',
```
Comment (goodboy) [inline tests/conftest.py:70]: We should probably have a CD process eventually that keeps us in sync with core fs.

**Comment by goodboy** [comment]
Hmm also, aren't we going to need a version pin in `setup.py` for 3.6?

Gonna be nice to get a new release out if we can get this all up and running again 🏄🏼

File: tests/conftest.py (hunk 2 above)
Comment (Kick1911) [reply tests/conftest.py:70]: Looks like the maintainer added a latest tag 4 months ago. I'll use that tag instead.

**Comment by Kick1911** [comment]
@goodboy Ok, I think it is all good now. All tests pass.
https://github.com/sudonum/switchio/runs/2484324517?check_suite_focus=true

File: tests/conftest.py [hunk 3]
```diff
@@ -67,7 +67,7 @@ def containers(request, confdir):
     if request.config.option.usedocker:
         docker = request.getfixturevalue('dockerctl')
         with docker.run(
-            'safarov/freeswitch:1.10.3',
+            'safarov/freeswitch:latest',
```
Comment (goodboy) [inline tests/conftest.py:66]: 🥳

File: tests/apps/test_measure.py [hunk 4]
```diff
@@ -271,6 +271,11 @@ def test_with_orig(get_orig, measure, storer):
 
     # configure max calls originated to length of of storer buffer
     cdr_storer = orig.measurers['CDR'].storer
+
+    if (cdr_storer.storetype.__name__ == 'CSVStore'
```
Comment (goodboy) [inline tests/apps/test_measure.py:275]: Not sure what's up here but we're skipping this measurement test for now since it's flaky in github actions.

**Comment by goodboy** [comment]
Created #74 to track otherwise I think this is ready to land.

@Kick1911 only last thing might be to change readme to say we're 3.6+ now yah?

File: tests/apps/test_measure.py (hunk 4 above)
Comment (Kick1911) [reply tests/apps/test_measure.py:275]: This was to skip that storertype, but not skip the other storertype. I didn't know if I should skip the test completely for all python 3.7+ or not.

**Comment by Kick1911** [comment]
> Created #74 to track otherwise I think this is ready to land.
> 
> @Kick1911 only last thing might be to change readme to say we're 3.6+ now yah?

Sure. Will do.

**Comment by Kick1911** [comment]
@goodboy Done

**Comment by goodboy** [comment]
@Kick1911 oh shoot we probably need to move the CI readme badge to GH actions as well.

Feel free to PR that in if you want but no presh.

**Comment by goodboy** [comment]
`master` run : https://github.com/friends-of-freeswitch/switchio/actions/runs/804780948

**Pull request merged**